src/evolutionary_keras/optimizers.py

```diff
@@ -216,7 +217,158 @@ def run_step(self, x, y):
+    """
+
+    def weights_per_layer(self):
+        """
+        The first values of 'self.length_flat_layer' is set to 0 which is helpful
+        in determining the range of weights in the function 'undo_flatten'. 
+        """
+        self.length_flat_layer = [0]
+        for layer in range(len(self.shape)):
+            flat_layer = np.reshape(self.weightscopy[layer].numpy(), [-1])
+            self.length_flat_layer.append(len(flat_layer))
+
+    """
+    'flatten' returns a 1 dimensional list of all weights in the keras model 
+    """
+
+    def flatten(self):
+        weights = self.weightscopy
+        flattened_weights = []
+        for layer in range(len(weights)):
+            flattened_weights.append(np.reshape(weights[layer].numpy(), [-1]))
+        flattened_weights = np.concatenate(flattened_weights)
+        return flattened_weights
+
+    """ 
+    'undo_flatten' does the inverse of 'flatten': it takes a 1 dimensional input
+    and returns a weight structure that can be loaded into the model. 
+    """
+
+    def undo_flatten(self, flattened_weights):
+        new_weights = []
+        for i in range(len(self.shape)):
+            flat_layer = flattened_weights[
+                self.length_flat_layer[i] : self.length_flat_layer[i]
+                + self.length_flat_layer[i + 1]
+            ]
+            new_weights.append(np.reshape(flat_layer, self.shape[i]))
+
+        ordered_names = [
+            weight.name for layer in self.model.layers for weight in layer.weights
+        ]
+
+        new_parent = deepcopy(self.model.get_weights())
+        i = 0
+        for weight in self.trainable_weights_names:
+            location_weight = ordered_names.index(weight)
+            new_parent[location_weight] = new_weights[i]
+            i += 1
+
+        return new_parent
+
+    """     
+    As one might have noticed, 'CMA' does not allow the user to set a number of
+    epochs, as this is dealth with by 'cma'. The default 'epochs' in EvolModel
+    is one, meaning 'run step' is only called once during training. 
+    """
+
+    def run_step(self, x, y):
+        """ Wrapper to the optimizer"""
+
+        """ 
+        Get the nubmer of weights in each keras layer 
+        """
+        self.weights_per_layer()
+
+        """ 
+        The function that 'cma' aims to minimize 
+        """
+
+        metricas = self.model.metrics_names
+
+        def minimizethis(flattened_weights):
+            weights = self.undo_flatten(flattened_weights)
+            self.model.set_weights(weights)
+            score = self.model.evaluate(x=x, y=y, verbose=0)
+            loss = parse_eval(score)
+            try:
+                history_data = dict(zip(metricas, score))
+            except TypeError as e:
+                if loss == score:
+                    score = [score, score]
+                    history_data = dict(zip(metricas, score))
+                else:
+                    raise e

```

Comment:
What is history data used here for? Maybe I’m missing something, but I think you can just return `loss`